eos/maths/legendre-polynomial-vector.hh

```diff
@@ -0,0 +1,267 @@
+
+            // Evaluate the vector of Legendre polynomials
+            std::array<double, order_ + 1> constexpr operator() (const double & z) const
+            {
+                std::array<double, order_ + 1> ret_vec;
+
+                if (order_ == 0)
+                {
+                    ret_vec[0] = 1;
+                }
+                else if (order_ == 1)
+                {
+                    ret_vec[0] = 1;
+                    ret_vec[1] = z;
+                }
+                else
+                {
+                    ret_vec[0] = 1;
+                    ret_vec[1] = z;
+                    for (unsigned i = 2 ; i <= order_ ; i++)
+                    {
+                        ret_vec[i] = (z * ret_vec[i - 1] * (2 * i - 1) - ret_vec[i - 2] * (i - 1)) / i;
+                    }
+                }
+                return ret_vec;
+            }
+
+            // Return zeros and compute Gauss-Legendre weights
+            void gauss_legendre(std::array<double, order_> & zeros, std::array<double, order_> & weights)
+            {
+                std::vector<double> zerosp = boost::math::legendre_p_zeros<double>(order_);
+                std::array<double, order_ / 2 + order_ % 2> weightsp;
+                const unsigned len = order_ / 2 + order_ % 2;
+
+                for (unsigned i = 0 ; i < len ; i++)
+                {
+                    double lp = boost::math::legendre_p<double>(order_ + 1, zerosp[i]);
+                    weightsp[i] = 2 * (1 - zerosp[i] * zerosp[i]) / ((order_ + 1) * (order_ + 1) * lp * lp);
+                }
+
+                // Flip order for first half
+                for (unsigned i = 0 ; i < len ; i++)
+                {
+                    zeros[i] = -1 * zerosp[len - 1 - i];
+                    weights[i] = weightsp[len - 1 - i];
+                }
+
+                // Double them, take care in case of odd order_
+                if (order_ % 2 == 1)
+                {
+                    for (unsigned i = 1 ; i < len ; i++)
+                    {
+                        zeros[len + i - 1] = zerosp[i];
+                        weights[len + i - 1] = weightsp[i];
+                    }
+                }
+                else
+                {
+                    for (unsigned i = 0 ; i < len ; i++)
+                    {
+                        zeros[len + i] = zerosp[i];
+                        weights[len + i] = weightsp[i];
+                    }
+                }
+            }
+    };
+
+    /*
+     * Representation of a vector of the real part of the associated Legendre functions for m = 0.
+     * Implementation is based on Zhang, Shanjie and Jin, Jianming. “Computation of Special Functions”, John Wiley and Sons, 1996.
+     *
+     */
+    template <unsigned order_>
+    class LegendreReQVector
+    {
+        private:
+            const double _eps;
+            const unsigned _cut;
+
+        public:
+            LegendreReQVector() :
+                _eps(1e-14),
```

Comment:
A comment on both values would be nice, to judge why they have been chosen like this.